autopilot/contractor.go

```diff
@@ -689,16 +716,18 @@ func (c *contractor) runContractRefreshes(ctx context.Context, w Worker, budget
 			break
 		}
 
-		contract, proceed, err := c.refreshContract(ctx, w, ci, budget, renterAddress)
+		contract, proceed, discard, err := c.refreshContract(ctx, w, ci, budget, renterAddress)
 		if err == nil {
 			refreshed = append(refreshed, contract)
+		} else if !discard {
+			discarded = append(discarded, contract.ID)
```

Comment:
You append to `discarded` when `!discard`?

Reply:
oversight when refactoring, flipped the logic at one point 😅  used to be named differently